File: app/src/main/java/com/b_lam/resplash/data/model/Me.java
package com.b_lam.resplash.data.model;

/**
 * Me
 **/

public class Me {

    /**
     "id": "pXhwzz1JtQU",
     "updated_at": "2016-07-10T11:00:01-05:00",
     "username": "jimmyexample",
     "first_name": "James",
     "last_name": "Example",
     "twitter_username": "jimmy",
     "portfolio_url": null,
     "bio": "The user's bio",
     "location": "Montreal, Qc",
     "total_likes": 20,
     "total_photos": 10,
     "total_collections": 5,
     "followed_by_user": false,
     "downloads": 4321,
     "uploads_remaining": 4,
     "instagram_username": "james-example",
     "email": "jim@example.com",
     "links": {
        "self": "https://api.unsplash.com/users/jimmyexample",
        "html": "https://unsplash.com/jimmyexample",
        "photos": "https://api.unsplash.com/users/jimmyexample/photos",
        "likes": "https://api.unsplash.com/users/jimmyexample/likes",
        "portfolio": "https://api.unsplash.com/users/jimmyexample/portfolio"
     }
     */

    public String id;
    public String updated_at;
    public String username;
    public String first_name;
    public String last_name;
    public String twitter_username;
    public String portfolio_url;
    public String bio;
    public String location;
    public int total_likes;
    public int total_photos;
    public int total_collections;
    public boolean followed_by_user;
    public int downloads;
    public int uploads_remaining;
    public String instagram_username;
    public String email;

    public UserLinks links;
}

File: app/src/main/java/com/b_lam/resplash/data/user/model/Me.kt
package com.b_lam.resplash.data.user.model

import com.b_lam.resplash.data.photo.model.Photo
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Me(
    val id: String,
    val updated_at: String,
    val username: String?,
    val first_name: String?,
    val last_name: String?,
    val twitter_username: String?,
    val portfolio_url: String?,
    val bio: String?,
    val location: String?,
    val links: Links?,
    val profile_image: ProfileImage?,
    val instagram_username: String?,
    val total_likes: Int?,
    val total_photos: Int?,
    val total_collections: Int?,
    val photos: List<Photo>?,
    val followed_by_user: Boolean?,
    val followers_count: Int?,
    val following_count: Int?,
    val downloads: Int?,
    val uploads_remaining: Int?,
    val email: String?
)
